src/Test/PHPUnit/AbstractCacheConstraint.php

```diff
@@ -41,17 +43,25 @@ public function __construct($header = null)
 
     /**
      * {@inheritdoc}
+     *
+     * @param Response $other The guzzle response object
      */
     protected function matches($other)
     {
+        if (!$other instanceof ResponseInterface) {
+            throw new \RuntimeException(sprintf('Expected a GuzzleHttp\Psr7\Response but got %s', get_class($other)));
+        }
         if (!$other->hasHeader($this->header)) {
-            throw new \RuntimeException(
-                sprintf(
-                    'Response has no "%s" header. Configure your caching proxy '
-                    . 'to set the header with cache hit/miss status.',
-                    $this->header
-                )
+            $message = sprintf(
+                'Response has no "%s" header. Configure your caching proxy '
+                . 'to set the header with cache hit/miss status.',
+                $this->header
             );
+            if (200 !== $other->getStatusCode()) {
+                $message .= sprintf("\nStatus code of response is %s.", $other->getStatusCode());
```

Comment:
and an error inside the kernel could abort the request. this is an additional hint if we expect a successful response but did not get one